pkg/detector/detector.go

```diff
@@ -986,6 +986,10 @@ func (d *ResourceDetector) HandleClusterPropagationPolicyDeletion(policyName str
 // And then check if object in waiting list matches the policy, if yes remove the object
 // from waiting list and throw the object to it's reconcile queue. If not, do nothing.
 func (d *ResourceDetector) HandlePropagationPolicyCreationOrUpdate(policy *policyv1alpha1.PropagationPolicy) error {
+	if err := d.handlePropagationPolicyPreemption(policy); err != nil {
```

Comment:
good advice.

Reply:
> it may change the claimed relationship between resource templates and (cluster or non-cluster)propagation policies.

I don't get it, does this really need to be considered at the beginning of the function?
Can you explain more?

Reply:
emm, I just follow the design flow. and it's ok to consider the preemption process at the beginning of the function or not, but since we've agreed on the design before, we'd better stick to the design.
https://github.com/karmada-io/karmada/blob/master/docs/proposals/scheduling/policy-preemption/preemption_process.PNG

Reply:
Consider that preemption is not the main process and will not influence it, I'd like to put it in the end. Because it will not affect the main process if there is an error in the preemption process.

Reply:
it looks good to me, could you update the [the design flow](https://github.com/karmada-io/karmada/blob/master/docs/proposals/scheduling/policy-preemption/preemption_process.PNG) synchronously?

Reply:
and maybe we can shorten the preemption process path. we don't need to overwrite the policy labels on resource template but apply it directly.